script/transform-approved-books-data.ts

```diff
@@ -0,0 +1,168 @@
+  return isApprovedVersionRepo(versionData) && versionData.repo === repoData.repo;
+};
+
+const matchCollectionVersion = (collectionData: ApprovedCollection) =>
+  (versionData: ApprovedVersionCollection | ApprovedVersionRepo): versionData is ApprovedVersionCollection => {
+  return isApprovedVersionCollection(versionData) && versionData.collection_id === collectionData.collection_id;
+};
+
+const getDesiredVersion = (
+  approvedVersions: Array<ApprovedVersionCollection | ApprovedVersionRepo>,
+  colOrRepo: ApprovedRepo | ApprovedCollection
+): string | undefined => {
+  const isCollection = isApprovedCollection(colOrRepo);
+
+  const filter = isCollection
+    ? matchCollectionVersion(colOrRepo as ApprovedCollection)
+    : matchRepoVersion(colOrRepo as ApprovedRepo);
+
+  const transformVersion = (versionData: ApprovedVersionCollection | ApprovedVersionRepo) => {
+    // books from collections have versions starting with 1.
+    return isCollection ? versionData.content_version.slice(2) : versionData.content_version;
+  };
+
+  const versions = approvedVersions
+    // only versions for current repo / collection
+    .filter(filter)
+    .map(transformVersion)
+    // sort and revert so results are descending
+    .sort()
+    .reverse();
+
+  // in general we want the most recent one, in the future we'll have to filter this by which ones
+  // are available for the recipe code version REX supports
+  return versions[0];
+};
+
+const { data } = argv as { data?: string };
+
+const transformData = () => {
+  if (!data) {
+    throw new Error('data argument is missing');
+  }
+
+  let parsed: ApprovedBooksAndVersions;
+  try {
+    parsed = JSON.parse(data);
+    if (!isApprovedBooksAndVersions(parsed)) {
+      throw new Error('Data is valid JSON, but has wrong structure. See ApprovedBooksAndVersions interface.');
+    }
+  } catch (e) {
+    throw new Error(`Error while parsing data. Details: ${e}`);
+  }
+
+  const results: { [key: string]: string } = {};
+
+  const { approved_books, approved_versions } = parsed;
+
+  for (const collectionOrRepo of approved_books) {
+    const books = collectionOrRepo.books;
+    const desiredVersion = getDesiredVersion(approved_versions, collectionOrRepo);
+
+    if (!desiredVersion) {
+      throw new Error(`Couldn't find version for ${
+        isApprovedRepo(collectionOrRepo)
+          ? `repo ${collectionOrRepo.repo}`
+          : `collection ${collectionOrRepo.collection_id}`
+      }`);
+    }
+
+    for (const book of books) {
+      if (!configuredBooks[book.uuid] || configuredBooks[book.uuid].defaultVersion !== desiredVersion) {
+        results[book.uuid] = desiredVersion;
+      }
+    }
+  }
+
+  // this script is used in a update-content/script.bash
+  // so we write results to the terminal to assign them to a variable
+  // tslint:disable-next-line: no-console
+  console.log(JSON.stringify(results));
```

Comment:
`console.log` includes formatting, you're better off calling `process.stdout.write` directly